qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
As I indicated in this thread, one possible solution is to have global IDs which are the aggregate, as long as we do not have some places in code which expect "sample_id" and other places which expect "study_id.sample_id". The reason that solution is not ideal is that it denormalizes as well as encourages developers to think of the sample ID as more than just an ID. This method is an example of where we are violating the concept of a sample ID being just an ID. I do not know if there are other places in qiita impacted, but I would not be surprised if there are and I definitely would not be surprised if methods that do this get merged in in the future.

I've stated my concerns over these approaches. If no one else is concerned, then we should just drop it.